`to_datetime`  falsely rejects `NDArray[float64]`

At runtime this is not a problem; `pd.to_datetime(np.linspace(0, 1, 1))` works fine. But the following `.pyi` repro shows that type-checkers reject this:

```pyi
import numpy as np
import numpy.typing as npt
import pandas as pd

a: npt.NDArray[np.float64]
pd.to_datetime(a)  # ❌ 
```

<details><summary>Output</summary>
<p>


mypy:

```
demo.pyi:6: error: Argument 1 to "to_datetime" has incompatible type "ndarray[tuple[Any, ...], dtype[float64]]"; expected "Sequence[float | date] | list[str] | tuple[float | str | date, ...] | ndarray[tuple[Any, ...], dtype[datetime64[date | int | None]]] | ndarray[tuple[Any, ...], dtype[str_]] | ndarray[tuple[Any, ...], dtype[signedinteger[_64Bit]]] | Index[Any] | ExtensionArray"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```

basedpyright:

```
demo.pyi
  demo.pyi:6:1 - error: No overloads for "to_datetime" match the provided arguments (reportCallIssue)
  demo.pyi:6:16 - error: Argument of type "NDArray[float64]" cannot be assigned to parameter "arg" of type "Sequence[float | date] | list[str] | tuple[float | str | date, ...] | np_ndarray_dt[tuple[Any, ...]] | np_ndarray_str[tuple[Any, ...]] | np_ndarray_int64[tuple[Any, ...]] | Index[Any] | ExtensionArray" in function "to_datetime"
    Type "NDArray[float64]" is not assignable to type "Sequence[float | date] | list[str] | tuple[float | str | date, ...] | np_ndarray_dt[tuple[Any, ...]] | np_ndarray_str[tuple[Any, ...]] | np_ndarray_int64[tuple[Any, ...]] | Index[Any] | ExtensionArray"
      "ndarray[_AnyShape, dtype[float64]]" is not assignable to "Sequence[float | date]"
      "ndarray[_AnyShape, dtype[float64]]" is not assignable to "list[str]"
      "ndarray[_AnyShape, dtype[float64]]" is not assignable to "tuple[float | str | date, ...]"
      "ndarray[_AnyShape, dtype[float64]]" is not assignable to "ndarray[tuple[Any, ...], dtype[datetime64[date | int | None]]]"
        Type parameter "_DTypeT_co@ndarray" is covariant, but "dtype[float64]" is not a subtype of "dtype[datetime64[date | int | None]]"
          "dtype[float64]" is not assignable to "dtype[datetime64[date | int | None]]"
            Type parameter "_ScalarT_co@dtype" is covariant, but "float64" is not a subtype of "datetime64[date | int | None]"
    ... (reportArgumentType)
2 errors, 0 warnings, 0 notes
```

pyrefly

```
ERROR No matching overload found for function `pandas.core.tools.datetimes.to_datetime` called with arguments: (ndarray[tuple[Any, ...], dtype[float64]]) [no-matching-overload]
 --> demo.pyi:6:15
  |
6 | pd.to_datetime(a)
  |               ^^^
  |
  Possible overloads:
  (arg: DatetimeScalar, errors: Literal['raise'] = 'raise', dayfirst: bool = False, yearfirst: bool = False, utc: bool = False, format: str | None = None, exact: bool = True, unit: Literal['D', 'ms', 'ns', 's', 'us'] | None = None, origin: Timestamp | date | datetime | datetime64[date | int | None] | float | integer[Any] | str = 'unix', cache: bool = True) -> Timestamp [closest match]
  (arg: DatetimeScalar, errors: Literal['coerce'], dayfirst: bool = False, yearfirst: bool = False, utc: bool = False, format: str | None = None, exact: bool = True, unit: Literal['D', 'ms', 'ns', 's', 'us'] | None = None, origin: Timestamp | date | datetime | datetime64[date | int | None] | float | integer[Any] | str = 'unix', cache: bool = True) -> NaTType | Timestamp
  (arg: DataFrame | Series[Any] | FulldatetimeDict, errors: RaiseCoerce = 'raise', dayfirst: bool = False, yearfirst: bool = False, utc: bool = False, format: str | None = None, exact: bool = True, unit: Literal['D', 'ms', 'ns', 's', 'us'] | None = None, origin: Timestamp | date | datetime | datetime64[date | int | None] | float | integer[Any] | str = 'unix', cache: bool = True) -> Series[Timestamp]
  (arg: ExtensionArray | Index[Any] | Sequence[date | float] | list[str] | ndarray[tuple[Any, ...], dtype[datetime64[date | int | None]]] | ndarray[tuple[Any, ...], dtype[signedinteger[_64Bit]]] | ndarray[tuple[Any, ...], dtype[str_]] | tuple[date | float | str, ...], errors: RaiseCoerce = 'raise', dayfirst: bool = False, yearfirst: bool = False, utc: bool = False, format: str | None = None, exact: bool = True, unit: Literal['D', 'ms', 'ns', 's', 'us'] | None = None, origin: Timestamp | date | datetime | datetime64[date | int | None] | float | integer[Any] | str = 'unix', cache: bool = True) -> DatetimeIndex
 INFO 1 error
```

ty:

```
error[no-matching-overload]: No overload of function `to_datetime` matches arguments
 --> demo.pyi:6:1
  |
5 | a: npt.NDArray[np.float64]
6 | pd.to_datetime(a)
  | ^^^^^^^^^^^^^^^^^
  |
info: First overload defined here
  --> .venv/lib/python3.14/site-packages/pandas-stubs/core/tools/datetimes.pyi:61:5
   |
60 |   @overload
61 |   def to_datetime(
   |  _____^
62 | |     arg: DatetimeScalar,
63 | |     errors: Literal["raise"] = "raise",
64 | |     dayfirst: bool = False,
65 | |     yearfirst: bool = False,
66 | |     utc: bool = False,
67 | |     format: str | None = None,
68 | |     exact: bool = True,
69 | |     unit: Literal["D", "s", "ms", "us", "ns"] | None = None,
70 | |     origin: Literal["julian", "unix"] | TimestampConvertibleTypes = "unix",
71 | |     cache: bool = True,
72 | | ) -> Timestamp: ...
   | |______________^
73 |   @overload
74 |   def to_datetime(
   |
info: Possible overloads for function `to_datetime`:
info:   (arg: int | float | str | date | datetime64[date | int | None], errors: Literal["raise"] = "raise", dayfirst: bool = False, yearfirst: bool = False, utc: bool = False, format: str | None = None, exact: bool = True, unit: Literal["D", "s", "ms", "us", "ns"] | None = None, origin: str | date | datetime64[date | int | None] | ... omitted 3 union elements = "unix", cache: bool = True) -> Timestamp
info:   (arg: int | float | str | date | datetime64[date | int | None], errors: Literal["coerce"], dayfirst: bool = False, yearfirst: bool = False, utc: bool = False, format: str | None = None, exact: bool = True, unit: Literal["D", "s", "ms", "us", "ns"] | None = None, origin: str | date | datetime64[date | int | None] | ... omitted 3 union elements = "unix", cache: bool = True) -> Timestamp | NaTType
info:   (arg: Series[Any] | FulldatetimeDict | DataFrame, errors: Literal["raise", "coerce"] = "raise", dayfirst: bool = False, yearfirst: bool = False, utc: bool = False, format: str | None = None, exact: bool = True, unit: Literal["D", "s", "ms", "us", "ns"] | None = None, origin: str | date | datetime64[date | int | None] | ... omitted 3 union elements = "unix", cache: bool = True) -> Series[Timestamp]
info:   (arg: Sequence[int | float | date] | list[str] | tuple[int | float | str | date, ...] | ... omitted 5 union elements, errors: Literal["raise", "coerce"] = "raise", dayfirst: bool = False, yearfirst: bool = False, utc: bool = False, format: str | None = None, exact: bool = True, unit: Literal["D", "s", "ms", "us", "ns"] | None = None, origin: str | date | datetime64[date | int | None] | ... omitted 3 union elements = "unix", cache: bool = True) -> DatetimeIndex
info: rule `no-matching-overload` is enabled by default

Found 1 diagnostic
```


</p>
</details> 

versions:

```
pandas==2.3.3
pandas-stubs==2.3.3.260113
```

--- a/pandas-stubs/core/tools/datetimes.pyi
+++ b/pandas-stubs/core/tools/datetimes.pyi
@@ -26,8 +26,9 @@ from pandas._typing import (
     DictConvertible,
     RaiseCoerce,
     TimestampConvertibleTypes,
+    np_ndarray_anyint,
     np_ndarray_dt,
-    np_ndarray_int64,
+    np_ndarray_float,
     np_ndarray_str,
 )
 
@@ -104,7 +105,8 @@ def to_datetime(
         | tuple[float | str | date, ...]
         | np_ndarray_dt
         | np_ndarray_str
-        | np_ndarray_int64
+        | np_ndarray_anyint
+        | np_ndarray_float
         | Index
         | ExtensionArray
     ),

--- a/tests/test_timefuncs.py
+++ b/tests/test_timefuncs.py
@@ -1499,6 +1499,22 @@ def test_to_datetime_array() -> None:
         assert_type(pd.to_datetime(np.array([1, 2, 3])), pd.DatetimeIndex),
         pd.DatetimeIndex,
     )
+    check(
+        assert_type(pd.to_datetime(np.array([1.0, 2.0, 3.0])), pd.DatetimeIndex),
+        pd.DatetimeIndex,
+    )
+    check(
+        assert_type(pd.to_datetime(np.array([1, 2, 3], np.uint8)), pd.DatetimeIndex),
+        pd.DatetimeIndex,
+    )
+    check(
+        assert_type(pd.to_datetime(np.array([1, 2, 3], np.int16)), pd.DatetimeIndex),
+        pd.DatetimeIndex,
+    )
+    check(
+        assert_type(pd.to_datetime(np.array([1, 2, 3], np.float32)), pd.DatetimeIndex),
+        pd.DatetimeIndex,
+    )
     pd.to_datetime(
         pd.Index([2451544.5, 2451545.5, 2451546.5]), unit="D", origin="julian"
     )

